resolver/dns_resolver_cache.ml

```diff
@@ -433,11 +450,24 @@ let handle_query t ~dnssec ~dnssec_ok ~rng ip_proto ts (qname, qtype) =
       else
         name, Fun.id
     in
-    let zone, name'', types, ip, t = resolve t ~dnssec ~rng ip_proto ts name' qtype in
-    let name'' = recover name'' in
-    Log.debug (fun m -> m "handle_query %a (%a) query %a, resolve zone %a query %a (%a), ip %a"
-                  Domain_name.pp qname Packet.Question.pp_qtype qtype
-                  Domain_name.pp name Domain_name.pp zone Domain_name.pp name''
-                  Fmt.(list ~sep:(any ", ") Packet.Question.pp_qtype) types
-                  Ipaddr.pp ip);
-    `Query (zone, (name'', types), ip), t
+    let actions = resolve t ~dnssec ip_proto ts name' qtype in
+    let up_to_three = pick_n rng 3 actions in
+    let ip1 = 4 - List.length actions in
```

Comment:
If I add a `max 1 ...` I don't get an exception from Randomconv, but I don't quite understand what ip1 and ip2 are for.
```suggestion
    let ip1 = max 1 (4 - List.length actions) in
```